> While connecting this with S9 miner with BraiinsOS I'm getting some erros and the errors are recurring at some interval. You need to check that.
> 
> ```
> [ERROR][01/16/26 23:34:49](miner_device) - Exception on /api/miners [GET]
> Traceback (most recent call last):
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/app.py", line 1511, in wsgi_app
>     response = self.full_dispatch_request()
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/app.py", line 919, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask_cors/extension.py", line 176, in wrapped_function
>     return cors_after_request(app.make_response(f(*args, **kwargs)))
>                                                 ^^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/app.py", line 917, in full_dispatch_request
>     rv = self.dispatch_request()
>          ^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/app.py", line 902, in dispatch_request
>     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/tests/miner_device.py", line 216, in get_miner_data
>     return jsonify(result)
>            ^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/json/__init__.py", line 170, in jsonify
>     return current_app.json.response(*args, **kwargs)  # type: ignore[return-value]
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/json/provider.py", line 214, in response
>     f"{self.dumps(obj, **dump_args)}\n", mimetype=self.mimetype
>        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/json/provider.py", line 179, in dumps
>     return json.dumps(obj, **kwargs)
>            ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/json/__init__.py", line 238, in dumps
>     **kw).encode(obj)
>           ^^^^^^^^^^^
>   File "/usr/lib/python3.12/json/encoder.py", line 200, in encode
>     chunks = self.iterencode(o, _one_shot=True)
>              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.12/json/encoder.py", line 258, in iterencode
>     return _iterencode(o, 0)
>            ^^^^^^^^^^^^^^^^^
>   File "/home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/flask/json/provider.py", line 121, in _default
>     raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
> TypeError: Object of type BraiinsOSError is not JSON serializable
> /home/zaidmstrr/projects/braidpool/.venv/lib/python3.12/site-packages/werkzeug/_internal.py:97 in _log
> [INFO][01/16/26 23:34:49](werkzeug) - 127.0.0.1 - - [16/Jan/2026 23:34:49] "GET /api/miners?ip=192.168.1.3 HTTP/1.1" 500 -
> ```

The issue was due to BraiinsOSError not being JSON serializable. I fixed it by converting error objects to strings and improving exception handling, which stopped the recurring 500 errors.

--- a/tests/miner_device.py
+++ b/tests/miner_device.py
@@ -222,3 +222,4 @@ def get_miner_data():
 
 if __name__ == '__main__':
     app.run(host='0.0.0.0', port=5001, debug=False)
+    

